metadata/src/main/java/com/redhat/lightblue/metadata/FieldConstraint.java

```diff
@@ -31,4 +31,10 @@
      * Determines if the constraint is valid for the given field type
      */
     boolean isValidForFieldType(Type fieldType);
+
+
+    /**
+     * Returns the description of the FieldConstraint (for documentation propose)
+     */
+    String getDescription();
```

Comment:
The documentation you  mean the getDescription method right? This method is just a way to use polymorphism so I can get the specific description of each constraint without casting or checking instanceof. The description is used to generate a user-friendly message in the json schema